File: server/utils/file.go
package utils

import "os"

func UpdateContent(content string, file string) (bool, error) {
	fileContent, err := os.ReadFile(file)
	if err == nil && string(fileContent) == content {
		return false, nil
	}
	err = os.WriteFile(file, []byte(content), 0644)
	if err != nil {
		return false, err
	}
	return true, nil
}

func FileExists(filename string) bool {
	_, err := os.Stat(filename)
	return !os.IsNotExist(err)
}

File: server/utils/file.py
import os

def update_content(content, file_path):
    if os.path.exists(file_path):
        with open(file_path, "r") as f:
            if f.read() == content:
                return False, None
    with open(file_path, "w") as f:
        f.write(content)
    return True, None

def file_exists(filename):
    return os.path.exists(filename)
